api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -433,4 +434,8 @@ message CertificateValidationContext {
   // Refer to the documentation for the specified validator. If you do not want a custom validation algorithm, do not set this field.
   // [#extension-category: envoy.tls.cert_validator]
   config.core.v3.TypedExtensionConfig custom_validator_config = 12;
+
+  // This option is true by default. If false, only the certificate at the end of the
+  // certificate chain will be subject to validation by CRL.
+  google.protobuf.BoolValue crl_verify_all = 14;
```

Comment:
I would potentially flip this, make it a basic type, and call it `only_verify_final_certificate_crl`. I think this would help make it a bit more clear and avoid using a WKT. WDYT?

Reply:
I agree. Also a field with default to false is more aligned with proto expectation. 
(Sorry for my bad naming in the original FR description :P )

Reply:
would be that leaf only? i.e. if you have an intermediate CA, not check all means intermediate CA will be checked but leaf only means no.

Reply:
```
X509_V_FLAG_CRL_CHECK enables CRL checking for the certificate chain leaf certificate. An error occurs if a suitable CRL cannot be found.
```

I spoke to Right. Yes, I think this means all present CRLs should be checked. Leaf is what is verified against, not the materials used as verification means.

i made another comment that we should have a test case for this.